core/src/wsv.rs

```diff
@@ -159,39 +155,58 @@ impl WorldStateView {
     ///
     /// # Errors
     /// Fails if there is no domain or account
-    pub fn account_assets(&self, id: &AccountId) -> Result<Vec<Asset>, QueryExecutionFail> {
-        self.map_account(id, |account| account.assets.values().cloned().collect())
+    pub fn account_assets(
+        &self,
+        id: &AccountId,
+    ) -> Result<impl ExactSizeIterator<Item = Asset> + '_, QueryExecutionFail> {
+        self.map_account(id, |account| account.assets.values().cloned())
     }
```

Comment:
This is a little out of line with the other similar functions. Why doesn't this return `ExactSizeIterator<Item = &Asset>` and allow the user to then apply `Iter::cloned` to them?

Reply:
I agree, there must have been some pesky lifetime issue, I'll try to return a reference